server/src/main/java/org/elasticsearch/index/codec/vectors/ES815BitFlatVectorsFormat.java

```diff
@@ -61,14 +62,14 @@ public String toString() {
         @Override
         public RandomVectorScorerSupplier getRandomVectorScorerSupplier(
             VectorSimilarityFunction vectorSimilarityFunction,
-            RandomAccessVectorValues randomAccessVectorValues
+            KnnVectorValues vectorValues
         ) throws IOException {
-            assert randomAccessVectorValues instanceof RandomAccessVectorValues.Bytes;
+            assert vectorValues instanceof ByteVectorValues;
             assert vectorSimilarityFunction == VectorSimilarityFunction.EUCLIDEAN;
-            if (randomAccessVectorValues instanceof RandomAccessVectorValues.Bytes randomAccessVectorValuesBytes) {
-                assert randomAccessVectorValues instanceof RandomAccessQuantizedByteVectorValues == false;
+            if (vectorValues instanceof ByteVectorValues byteVectorValues) {
```

Comment:
I wonder if we should call getEncoding here, but we still need to cast to ByteVectorValues, so it does not help much?

Reply:
yeah, we still need the cast anyway.